src/Euclidean.jl

```diff
@@ -170,20 +170,36 @@ end
 
 function basis(M::Euclidean{<:Tuple,F}, x, B::ArbitraryOrthonormalBasis) where F
     vecs = [_euclidean_basis_vector(x, i) for i in eachindex(x)]
-    return PrecomputedOrthonormalBasis(vecs)
+    if F == ℝ
```

Comment:
I'd prefer that this is implemented as an overload of `basis` rather than burying this check within a function.

Reply:
Ok, I'll change that.